src/EditorFeatures/Core/Wrapping/WrapItemsAction.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.Editor.Wrapping
+{
+    /// <summary>
+    /// Code action for actually wrapping items.  Provided as a special subclass because it will
+    /// also update the wrapping most-recently-used list when the code action is actually
+    /// invoked.
+    /// </summary>
+    internal class WrapItemsAction : DocumentChangeAction
+    {
+        // Keeps track of the invoked code actions.  That way we can prioritize those code actions 
+        // in the future since they're more likely the ones the user wants.  This is important as 
+        // we have 9 different code actions offered (3 major groups, with 3 actions per group).  
+        // It's likely the user will just pick from a few of these. So we'd like the ones they
+        // choose to be prioritized accordingly.
+        private static ImmutableArray<string> s_mruTitles = ImmutableArray<string>.Empty;
+
+        public string ParentTitle { get; }
+
+        public string SortTitle { get; }
+
+        public WrapItemsAction(string title, string parentTitle, Func<CancellationToken, Task<Document>> createChangedDocument)
+            : base(title, createChangedDocument)
+        {
+            ParentTitle = parentTitle;
+            SortTitle = parentTitle + "_" + title;
+        }
+
+        protected override Task<IEnumerable<CodeActionOperation>> ComputePreviewOperationsAsync(CancellationToken cancellationToken)
+        {
+            // For preview, we don't want to compute the normal operations.  Specifically, we don't
+            // want to compute the stateful operation that tracks which code action was triggered.
```

Comment:
❔ Are there tests covering this behavior specifically? It seems important since previews are frequently used.

Reply:
yes. i've tested this. not sure if i pushed those tests up yet.  have to go look at the state of my branch.  We def test the *state-less-ness* of things in tests.  But the stateful tests needed a bit of extra finagling.